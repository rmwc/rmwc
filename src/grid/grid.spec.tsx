import React from 'react';
import { render } from '@testing-library/react';
import { Grid, GridCell, GridRow } from './';

describe('Grid', () => {
  it('renders grid', () => {
    const grid = render(<Grid />);

    expect(grid.container.firstChild).toHaveClass('mdc-layout-grid');

    expect(grid.asFragment()).toMatchSnapshot();
  });

  it('renders gridCell', () => {
    const gridCell = render(<GridCell />);

    expect(gridCell.container.firstChild).toHaveClass('mdc-layout-grid__cell');

    expect(gridCell.asFragment()).toMatchSnapshot();
  });

  it('can be fixedColumnWidth', () => {
    const { asFragment } = render(<Grid fixedColumnWidth />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('automatically renders inner grid', () => {
    const { container } = render(<Grid />);
    expect(container.firstChild?.firstChild).toHaveClass(
      'mdc-layout-grid__inner'
    );
  });

  it('can have a custom GridRow component', () => {
    const { asFragment } = render(
      <Grid>
        <GridRow />
      </Grid>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('can be align', () => {
    const { asFragment } = render(
      <>
        <Grid align="left" />
        <Grid align="right" />
      </>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('GridCell span', () => {
    const { asFragment } = render(<GridCell span={4} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('GridCell phone', () => {
    const { asFragment } = render(<GridCell phone={4} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('GridCell tablet', () => {
    const { asFragment } = render(<GridCell tablet={4} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('GridCell desktop', () => {
    const { asFragment } = render(<GridCell desktop={4} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('GridCell order', () => {
    const { asFragment } = render(<GridCell order={4} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('GridCell align', () => {
    const { asFragment } = render(<GridCell align="top" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('can have custom classnames', () => {
    [Grid, GridCell].forEach((Component: React.ComponentType<any>) => {
      const { container } = render(
        <Component className={'my-custom-classname'} />
      );
      expect(container.firstChild).toHaveClass('my-custom-classname');
    });
  });
});
