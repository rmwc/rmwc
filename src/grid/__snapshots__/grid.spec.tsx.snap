// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid GridCell align 1`] = `
<DocumentFragment>
  <div
    class="mdc-layout-grid__cell mdc-layout-grid__cell--align-top"
  />
</DocumentFragment>
`;

exports[`Grid GridCell desktop 1`] = `
<DocumentFragment>
  <div
    class="mdc-layout-grid__cell mdc-layout-grid__cell--span-4-desktop"
  />
</DocumentFragment>
`;

exports[`Grid GridCell order 1`] = `
<DocumentFragment>
  <div
    class="mdc-layout-grid__cell mdc-layout-grid__cell--order-4"
  />
</DocumentFragment>
`;

exports[`Grid GridCell phone 1`] = `
<DocumentFragment>
  <div
    class="mdc-layout-grid__cell mdc-layout-grid__cell--span-4-phone"
  />
</DocumentFragment>
`;

exports[`Grid GridCell span 1`] = `
<DocumentFragment>
  <div
    class="mdc-layout-grid__cell mdc-layout-grid__cell--span-4"
  />
</DocumentFragment>
`;

exports[`Grid GridCell tablet 1`] = `
<DocumentFragment>
  <div
    class="mdc-layout-grid__cell mdc-layout-grid__cell--span-4-tablet"
  />
</DocumentFragment>
`;

exports[`Grid can be align 1`] = `
<DocumentFragment>
  <div
    class="mdc-layout-grid mdc-layout-grid--align-left"
  >
    <div
      class="mdc-layout-grid__inner"
    />
  </div>
  <div
    class="mdc-layout-grid mdc-layout-grid--align-right"
  >
    <div
      class="mdc-layout-grid__inner"
    />
  </div>
</DocumentFragment>
`;

exports[`Grid can be fixedColumnWidth 1`] = `
<DocumentFragment>
  <div
    class="mdc-layout-grid mdc-layout-grid--fixed-column-width"
  >
    <div
      class="mdc-layout-grid__inner"
    />
  </div>
</DocumentFragment>
`;

exports[`Grid can have a custom GridRow component 1`] = `
<DocumentFragment>
  <div
    class="mdc-layout-grid"
  >
    <div
      class="mdc-layout-grid__inner"
    />
  </div>
</DocumentFragment>
`;

exports[`Grid renders grid 1`] = `
<DocumentFragment>
  <div
    class="mdc-layout-grid"
  >
    <div
      class="mdc-layout-grid__inner"
    />
  </div>
</DocumentFragment>
`;

exports[`Grid renders gridCell 1`] = `
<DocumentFragment>
  <div
    class="mdc-layout-grid__cell"
  />
</DocumentFragment>
`;
